x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ManageOwnApiKeyClusterPrivilege.java

```diff
@@ -93,6 +94,13 @@ protected boolean extendedCheck(String action, TransportRequest request, Authent
                 }
             } else if (request instanceof final QueryApiKeyRequest queryApiKeyRequest) {
                 return queryApiKeyRequest.isFilterForCurrentUser();
+            } else if (request instanceof UpdateApiKeyRequest) {
+                // Note: this returns true even for requests authenticated via API keys.
```

Comment:
Wanted to check on this: does the reasoning here make sense?

Reply:
Trying to update the API key using the API key itself should be an authorization error, _not_ because of technical constraint. An API key is _not_ the owner of itself. An analogy is that a user cannot update its own roles. It is a common use case that an API key is created by a higher privileged user (e.g. a superuser), and gets handed to a different user (or application) to use. That different user should not be able to update the key using the key itself since that could lead to privilege escalation.

What we haven't decided yet is whether an API key can own its derived keys. Right now, derived keys are owned by no one. If we decide a derived key should be owned by the API key that creates it, we may need revise the logic here to allow it. But this does not change the fact that an API key does _not_ own itself (and can _never_ update itself).

If we want to consider all the nuisances, we should:
1. AuthZ (403) error if `isCurrentAuthenticationUsingSameApiKeyIdFromRequest == true`.
2. 400 error if the authentication is an API Key

But I think simply having 403 for both cases is easier (both explanation and implementation). It is also a better consistency if you consider the future bulk update API.

Reply:
We had a sync on this on slack and decided to return `true` here and go for a `400`:

Fundamentally, the limitation that only an owner can update an API key sits outside the authorization layer: there are other privileges like `all` or `manage_security` that, from an access perspective, would enable an API key to update itself. However, we still prevent the behavior because of the agreed upon semantics of the update API key operation, namely that only the owner *user* can directly (things like run-as are still possible) update an API key and no other entity.

As such, a `400` is the more appropriate response here.